src/main/webapp/components/api/cache-model.service.js

```diff
@@ -11,6 +11,7 @@ angular.module('managementConsole.api')
                 this.modelController = cluster.getModelController();
                 this.lastRefresh = null;
                 this.data = null;
+                this.show = true;
```

Comment:
This is attribute specific for view, not for data, so it does not feel completely right for CacheModel to know about it. Few ideas: rename it to `hide`, that way by default model is shown, and there is no need to set it here: it is better because now Model does not have any knowledge about view logic, but bad side that still remains is that view pollutes model with its own data.

Best approach would probably be the following: instead of using attribute that describes if node/cache is hidden or not, you could calculate it on the fly: in controller, instead of checking for `ng-show="node.show"`, you could do `ng-show=isNodeShown(node)`, where `isNodeShown` would be your function that checks based on filters values and node values if node should be shown or not. In that case you do not even need to use '$watch' to react to filter values changes. I would say this is certainly the best way regarding the beauty of logic and code, only argument against would be performance: if there are many nodes and caches (for example hundred and more) then it may affect performance, also performance may be affected if it is complicated to fetch cache/node stats (although I believe rest calls are cached right? Or do we expect these stats to change so we can not cache it?).